runtime/src/bank.rs

```diff
@@ -5034,6 +5034,29 @@ impl Bank {
         timings: &mut ExecuteTimings,
         log_messages_bytes_limit: Option<usize>,
     ) -> (Vec<TransactionCommitResult>, TransactionBalancesSet) {
+        self.load_execute_and_commit_transactions_with_pre_commit_callback(
+            batch,
+            max_age,
+            collect_balances,
+            recording_config,
+            timings,
+            log_messages_bytes_limit,
+            None::<fn() -> bool>,
+        )
+        .unwrap()
+    }
+
+    #[must_use]
+    pub fn load_execute_and_commit_transactions_with_pre_commit_callback(
+        &self,
+        batch: &TransactionBatch<impl TransactionWithMeta>,
+        max_age: usize,
+        collect_balances: bool,
+        recording_config: ExecutionRecordingConfig,
+        timings: &mut ExecuteTimings,
+        log_messages_bytes_limit: Option<usize>,
+        pre_commit_callback: Option<impl FnOnce() -> bool>,
+    ) -> Option<(Vec<TransactionCommitResult>, TransactionBalancesSet)> {
```

Comment:
This also seems like it should be a `Result` instead of an `Option`, because this **should** succeed and fails due to something going wrong.

Reply:
this is done: 3b852f6a948ba6c7ae36ecdab85696e8dc284023